File: api/join.py
import csv
import json
from datetime import datetime, timezone
from pathlib import Path

TELEGRAM_URL = "https://t.me/+0tgMl32I9foxMmNh"
FIELDS = ["submitted_at", "first_name", "last_name", "email", "business_name", "role"]
CSV_PATH = Path("/tmp/aiclub_members.csv")

def handler(request):
    if request.method != "POST":
        return {
            "statusCode": 405,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"ok": False, "error": "Method not allowed"})
        }

    try:
        payload = json.loads(request.body or "{}")
    except Exception:
        return {
            "statusCode": 400,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"ok": False, "error": "Invalid JSON"})
        }

    row = {
        "submitted_at": datetime.now(timezone.utc).isoformat(),
        "first_name": str(payload.get("first_name", "")).strip(),
        "last_name": str(payload.get("last_name", "")).strip(),
        "email": str(payload.get("email", "")).strip(),
        "business_name": str(payload.get("business_name", "")).strip(),
        "role": str(payload.get("role", "")).strip(),
    }

    if not all([row["first_name"], row["last_name"], row["email"], row["business_name"], row["role"]]):
        return {
            "statusCode": 400,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"ok": False, "error": "Missing required fields"})
        }

    write_header = not CSV_PATH.exists()
    with CSV_PATH.open("a", newline="", encoding="utf-8") as f:
        writer = csv.DictWriter(f, fieldnames=FIELDS)
        if write_header:
            writer.writeheader()
        writer.writerow(row)

    return {
        "statusCode": 200,
        "headers": {"Content-Type": "application/json"},
        "body": json.dumps({"ok": True, "redirect_url": TELEGRAM_URL})
    }

File: api/join.js
const fs = require('fs');
const path = require('path');

const TELEGRAM_URL = 'https://t.me/+0tgMl32I9foxMmNh';
const FIELDS = ['submitted_at', 'first_name', 'last_name', 'email', 'business_name', 'role'];
const CSV_PATH = '/tmp/aiclub_members.csv';

function parseBody(body) {
  if (!body) return {};
  if (typeof body === 'object') return body;
  if (typeof body === 'string') return JSON.parse(body);
  return {};
}

function csvEscape(value) {
  const text = String(value ?? '');
  if (/[",\n\r]/.test(text)) return `"${text.replace(/"/g, '""')}"`;
  return text;
}

module.exports = function handler(req, res) {
  if (req.method !== 'POST') {
    res.status(405).json({ ok: false, error: 'Method not allowed' });
    return;
  }

  let payload;
  try {
    payload = parseBody(req.body);
  } catch (_error) {
    res.status(400).json({ ok: false, error: 'Invalid JSON' });
    return;
  }

  const row = {
    submitted_at: new Date().toISOString(),
    first_name: String(payload.first_name || '').trim(),
    last_name: String(payload.last_name || '').trim(),
    email: String(payload.email || '').trim(),
    business_name: String(payload.business_name || '').trim(),
    role: String(payload.role || '').trim(),
  };

  if (!row.first_name || !row.last_name || !row.email || !row.role) {
    res.status(400).json({ ok: false, error: 'Missing required fields' });
    return;
  }

  const writeHeader = !fs.existsSync(CSV_PATH);
  const line = FIELDS.map((field) => csvEscape(row[field])).join(',') + '\n';
  if (writeHeader) {
    fs.writeFileSync(CSV_PATH, FIELDS.join(',') + '\n' + line, 'utf8');
  } else {
    fs.appendFileSync(CSV_PATH, line, 'utf8');
  }

  res.status(200).json({ ok: true, redirect_url: TELEGRAM_URL });
};
